api_tests/nodes/views/test_node_detail.py

```diff
@@ -845,24 +845,27 @@ def test_update_errors(
     def test_update_public_project_logged_in(
             self, app, user, title_new, description_new,
             category_new, project_public, url_public):
-        with assert_latest_log(NodeLog.UPDATED_FIELDS, project_public):
-            res = app.put_json_api(url_public, {
-                'data': {
-                    'id': project_public._id,
-                    'type': 'nodes',
-                    'attributes': {
-                        'title': title_new,
-                        'description': description_new,
-                        'category': category_new,
-                        'public': True
-                    }
+        res = app.put_json_api(url_public, {
+            'data': {
+                'id': project_public._id,
+                'type': 'nodes',
+                'attributes': {
+                    'title': title_new,
+                    'description': description_new,
+                    'category': category_new,
+                    'public': True
                 }
-            }, auth=user.auth)
-            assert res.status_code == 200
-            assert res.content_type == 'application/vnd.api+json'
-            assert res.json['data']['attributes']['title'] == title_new
-            assert res.json['data']['attributes']['description'] == description_new
-            assert res.json['data']['attributes']['category'] == category_new
+            }
+        }, auth=user.auth)
+        assert res.status_code == 200
+        assert res.content_type == 'application/vnd.api+json'
+        assert res.json['data']['attributes']['title'] == title_new
+        assert res.json['data']['attributes']['description'] == description_new
+        assert res.json['data']['attributes']['category'] == category_new
+        log_actions = [log.action for log in project_public.logs.all()]
```

Comment:
good suggestion, and you don't need `list` either!

Reply:
True, you don't need `list`, but I used it because of the several `foo in log_actions` checks -- without explicitly getting the queryset results, I think each is a separate DB query.

Reply:
Hmmm but I don't know which is slower -- one query to get a list of values, or three queries that return only boolean, no stored values. Probably doesn't much matter.